flink-runtime/src/main/java/org/apache/flink/runtime/metrics/MetricNames.java

```diff
@@ -50,6 +50,7 @@ private MetricNames() {}
     public static final String NUM_REGISTERED_TASK_MANAGERS = "numRegisteredTaskManagers";
 
     public static final String NUM_RESTARTS = "numRestarts";
+    public static final String NUM_JOB_FAILURE = "numJobFailure";
```

Comment:
I'm not sure if our users will be confused if there are two similar metrics: `numRestarts` and `numJobFailure`.
What's the exact difference between them?

I guess the number of failures can be higher than the number of restarts (local failures, ignored restarts) ... but I fear a user just looking at the available metrics might pick the first one available.

Reply:
Added comment to make it clearer for users.